src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -1157,6 +1163,9 @@ async def setup_pex_process(request: PexProcess, pex_environment: PexEnvironment
     complete_pex_env = pex_environment.in_sandbox(working_directory=request.working_directory)
     argv = complete_pex_env.create_argv(pex.name, *request.argv)
     env = {
+        # Set this in case this PEX was built with --no-pre-install-wheels, and thus parallelising
+        # the install on cold boot is handy.
+        "PEX_MAX_INSTALL_JOBS": str(request.concurrency_available),
```

Comment:
I think this is what you were getting at in your notes, but as a rough estimate if I have 16 cores this will potentially thrash into 16 different cache entries, right?

Reply:
Yeah, in theory... depending on what else is running at the same time.

However, on closer examination prompted by your comment: I think I did this wrong:

- `concurrency_available` is configuration to tell the command runner how much concurrency would be _useful_ for the process (e.g. "I'm processing 34 files, and could do so in parallel" => concurrency available = 34), not necessary how much it is actually allocated.
- That allocation is communicated to the process by replacing the `{pants_concurrency}` template in its argv (to be used like [`"some-process", ..., "--jobs={pants_concurrency}"]` or similar).
- This replacement currently only seems to work on the argv, not env vars.
 
For myself, relevant code (and surrounds), of the `concurrency_available` request flowing through to the replacement:

https://github.com/pantsbuild/pants/blob/b7b0e9ca349bc15c619c62c4ef9ba7787bf2b9d9/src/rust/engine/process_execution/src/bounded.rs#L75-L117

So, I'm gonna revert this part of the change. Thanks. 

https://github.com/pantsbuild/pants/pull/20795